### PR TITLE
JENKINS/GTEST: Test 'make distcheck' and fix gtest's Makefile.am

### PR DESCRIPTION
**Pull request by alinask**

### CODE REVIEW

File: contrib/test_jenkins.sh [hunk 1]
```diff
@@ -536,6 +536,27 @@ check_inst_headers() {
...
+	$MAKEP
+	$MAKEP distcheck
+
+	rm -rf ucx_new
+	mkdir ucx_new
+	tar -zxf $(ls -t ucx*.tar.gz | head -1) -C ucx_new
+	cd ucx_new/ucx*
+	./contrib/configure-release --enable-gtest --prefix=$PWD/install
+	$MAKEP
+	$MAKEP install
```
[the remark this answers is not in the document]
Comment (alinask) [reply contrib/test_jenkins.sh]: actually, in mtt, i saw it fail on the install phase - 
http://e2e-gw.mellanox.com:4080/mnt/lustre/users/mtt/scratch/ucx_ompi/20190610_104519_12943_388169_clx-orion-022/html/test_stdout_W8e5CO.txt
(Mellanox intrernal link)

[the remark this answers is not in the document]
Comment (yosefe) [reply contrib/test_jenkins.sh]: pls check if distchek already covers make+make install as needed, and if yes let's remove the 2nd step

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/10624/ for details (Mellanox internal link).

File: contrib/test_jenkins.sh (hunk 1 above)
[the remark this answers is not in the document]
Comment (alinask) [reply contrib/test_jenkins.sh]: @yosefe By '2nd step' you mean lines 548-555 ?
If I remove these lines, and remove the addition of the missing .h file from the Makefile.am, then the distcheck in line 546 passes:
```
ucx-1.7.0 archives ready for distribution:
ucx-1.7.0.tar.gz
```

Only if I keep these lines, the issue reproduces (after the dustribution is created) :
```
  CXX      uct/ib/gtest-test_ib.o
uct/ib/test_ib.cc:7:28: fatal error: uct/ib/test_ib.h: No such file or directory
 #include <uct/ib/test_ib.h>
                            ^
compilation terminated.
make[3]: *** [uct/ib/gtest-test_ib.o] Error 1
```

Here is what distcheck includes:

Automake also generates a distcheck rule that can be of help to ensure that a given distribution will actually work. Simplifying a bit, we can say this rule first makes a distribution, and then, operating from it, takes the following steps:

- tries to do a VPATH build (see VPATH Builds), with the srcdir and all its content made read-only;
- runs the test suite (with make check) on this fresh build;
- installs the package in a temporary directory (with make install), and tries runs the test suite on the resulting installation (with make installcheck);
- checks that the package can be correctly uninstalled (by make uninstall) and cleaned (by make distclean);
- finally, makes another tarball to ensure the distribution is self-contained.

https://www.gnu.org/software/automake/manual/html_node/Checking-the-Distribution.html

According to the third bullet, make+make install are included inside distcheck.

[the remark this answers is not in the document]
Comment (yosefe) [reply contrib/test_jenkins.sh]: pls use DISTCHECK_CONFIGURE_FLAGS

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/10635/ for details (Mellanox internal link).

**Comment by alinask** [comment]
@yosefe now that gtest compiles on the VMs in jenkins (due to this PR), there are compilation errors.
one of them is 
```
[2019-06-13T11:45:11.491Z] In file included from /usr/include/x86_64-linux-gnu/sys/poll.h:76:0,

[2019-06-13T11:45:11.491Z]                  from /usr/include/poll.h:1,

[2019-06-13T11:45:11.491Z]                  from ../../../../test/gtest/uct/ib/test_ib.h:10,

[2019-06-13T11:45:11.491Z]                  from ../../../../test/gtest/uct/ib/test_ib.cc:7:

[2019-06-13T11:45:11.491Z] In function ‘int poll(pollfd*, nfds_t, int)’,

[2019-06-13T11:45:11.491Z]     inlined from ‘virtual void test_uct_event_ib::init()’ at ../../../../test/gtest/uct/ib/test_ib.cc:365:9:

[2019-06-13T11:45:11.491Z] /usr/include/x86_64-linux-gnu/bits/poll2.h:43:65: error: call to ‘__poll_chk_warn’ declared with attribute warning: poll called with fds buffer too small file nfds entries [-Werror]

[2019-06-13T11:45:11.491Z]   return __poll_chk_warn (__fds, __nfds, __timeout, __bos (__fds));

[2019-06-13T11:45:11.491Z]                                                                  ^

[2019-06-13T11:45:11.491Z] cc1plus: all warnings being treated as errors
```

which seems to be a compiler bug: 
https://gcc.gnu.org/bugzilla/show_bug.cgi?id=61886

maybe exclude this (make distcheck) check depending on the gcc version then?

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/10653/ for details (Mellanox internal link).

**Comment by alinask** [comment]
bot:mlx:retest

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/10654/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
<details><summary>Mellanox CI: <b>PASSED</b#  > on 33 workers (click for details)</summary>

**Note:** the logs will be deleted after 24-Jun-2019

Agent/Stage | Status
-- | --
[_main](https://gist.githubusercontent.com/mellanox-github/bcc6a4599067962b40ac2cbbe9c57c31/raw/354fa2826512e5527274cde98ace787b56b547f5/_main) | :heavy_check_mark: SUCCESS
[hpc-arm-cavium-jenkins_W0](https://gist.githubusercontent.com/mellanox-github/bcc6a4599067962b40ac2cbbe9c57c31/raw/d5a11ee3c0014d0e51b172c74ce85926977e966b/hpc-arm-cavium-jenkins_W0) | :heavy_check_mark: SUCCESS
[hpc-arm-cavium-jenkins_W1](https://gist.githubusercontent.com/mellanox-github/bcc6a4599067962b40ac2cbbe9c57c31/raw/7be6f8c6b071de06c6cd14e3c8f78526d18a0071/hpc-arm-cavium-jenkins_W1) | :heavy_check_mark: SUCCESS
[hpc-arm-cavium-jenkins_W2](https://gist.githubusercontent.com/mellanox-github/bcc6a4599067962b40ac2cbbe9c57c31/raw/cb1a2ac8909b9f8c3888dfaec2d5cd6aad15bbab/hpc-arm-cavium-jenkins_W2) | :heavy_check_mark: SUCCESS
[hpc-arm-cavium-jenkins_W3](https://gist.githubusercontent.com/mellanox-github/bcc6a4599067962b40ac2cbbe9c57c31/raw/caae5d132547cce5ac2aeba53ce9fc004e25851f/hpc-arm-cavium-jenkins_W3) | :heavy_check_mark: SUCCESS
[hpc-arm-hwi-jenkins_W0](https://gist.githubusercontent.com/mellanox-github/bcc6a4599067962b40ac2cbbe9c57c31/raw/720ef7050a1fad4112389c421b8cfe3be7ee9f69/hpc-arm-hwi-jenkins_W0) | :heavy_check_mark: SUCCESS
[hpc-arm-hwi-jenkins_W1](https://gist.githubusercontent.com/mellanox-github/bcc6a4599067962b40ac2cbbe9c57c31/raw/fba08ac3043679f7a98367154f7b3b3083f1eeb4/hpc-arm-hwi-jenkins_W1) | :heavy_check_mark: SUCCESS
[hpc-arm-hwi-jenkins_W2](https://gist.githubusercontent.com/mellanox-github/bcc6a4599067962b40ac2cbbe9c57c31/raw/07c16afc01d6df12a8c356e8c758c36bfff892e0/hpc-arm-hwi-jenkins_W2) | :heavy_check_mark: SUCCESS
[hpc-arm-hwi-jenkins_W3](https://gist.githubusercontent.com/mellanox-github/bcc6a4599067962b40ac2cbbe9c57c31/raw/930102954c02e9d281699331795d27a1389fe9f4/hpc-arm-hwi-jenkins_W3) | :heavy_check_mark: SUCCESS
[hpc-test-node-upstream_W0](https://gist.githubusercontent.com/mellanox-github/bcc6a4599067962b40ac2cbbe9c57c31/raw/ee914dc70f3622b15f64855034052baacd4ff2d4/hpc-test-node-upstream_W0) | :heavy_check_mark: SUCCESS
[hpc-test-node-upstream_W1](https://gist.githubusercontent.com/mellanox-github/bcc6a4599067962b40ac2cbbe9c57c31/raw/07b17c907661b1b60d78fb0260dd20e6567d76d6/hpc-test-node-upstream_W1) | :heavy_check_mark: SUCCESS
[hpc-test-node-upstream_W2](https://gist.githubusercontent.com/mellanox-github/bcc6a4599067962b40ac2cbbe9c57c31/raw/128364ea2c6571dc099262786596ab41f8a4af44/hpc-test-node-upstream_W2) | :heavy_check_mark: SUCCESS
[hpc-test-node-upstream_W3](https://gist.githubusercontent.com/mellanox-github/bcc6a4599067962b40ac2cbbe9c57c31/raw/5061498bb47265c0f5ee2f50ee7d0ef76e28b43f/hpc-test-node-upstream_W3) | :heavy_check_mark: SUCCESS
[hpc-test-node2_W0](https://gist.githubusercontent.com/mellanox-github/bcc6a4599067962b40ac2cbbe9c57c31/raw/89268547f5c7209b85f06c4bb60fa660bbfa8b89/hpc-test-node2_W0) | :heavy_check_mark: SUCCESS
[hpc-test-node2_W1](https://gist.githubusercontent.com/mellanox-github/bcc6a4599067962b40ac2cbbe9c57c31/raw/5aa82161fe55833421b68ba5084b1cd171c8e0ca/hpc-test-node2_W1) | :heavy_check_mark: SUCCESS
[hpc-test-node2_W2](https://gist.githubusercontent.com/mellanox-github/bcc6a4599067962b40ac2cbbe9c57c31/raw/ee7089014ad431ad662e643785d3a379b04cca79/hpc-test-node2_W2) | :heavy_check_mark: SUCCESS
[hpc-test-node2_W3](https://gist.githubusercontent.com/mellanox-github/bcc6a4599067962b40ac2cbbe9c57c31/raw/d3b16a7a11e0a970b24fe0e1f89bae22afedec15/hpc-test-node2_W3) | :heavy_check_mark: SUCCESS
[hpc-test-node3_W0](https://gist.githubusercontent.com/mellanox-github/bcc6a4599067962b40ac2cbbe9c57c31/raw/6a6573dc2a468534a15ad745f11aaf56f15e93c7/hpc-test-node3_W0) | :heavy_check_mark: SUCCESS
[hpc-test-node3_W1](https://gist.githubusercontent.com/mellanox-github/bcc6a4599067962b40ac2cbbe9c57c31/raw/2bdd059a9cc7d406bdb3c7958c6e1cef85d7631d/hpc-test-node3_W1) | :heavy_check_mark: SUCCESS
[hpc-test-node3_W2](https://gist.githubusercontent.com/mellanox-github/bcc6a4599067962b40ac2cbbe9c57c31/raw/f4b42b19e2939365e8b0b5e62fd865ab861fc085/hpc-test-node3_W2) | :heavy_check_mark: SUCCESS
[hpc-test-node3_W3](https://gist.githubusercontent.com/mellanox-github/bcc6a4599067962b40ac2cbbe9c57c31/raw/6def79dbf8e954023513e4b2bef8219c58ef84ea/hpc-test-node3_W3) | :heavy_check_mark: SUCCESS
[hpc-test-node_W0](https://gist.githubusercontent.com/mellanox-github/bcc6a4599067962b40ac2cbbe9c57c31/raw/ad68fba231eb00229d849af5c001a8a42782b28e/hpc-test-node_W0) | :heavy_check_mark: SUCCESS
[hpc-test-node_W1](https://gist.githubusercontent.com/mellanox-github/bcc6a4599067962b40ac2cbbe9c57c31/raw/b267305fa7e91f9eddd643292d4af476edf9cd69/hpc-test-node_W1) | :heavy_check_mark: SUCCESS
[hpc-test-node_W2](https://gist.githubusercontent.com/mellanox-github/bcc6a4599067962b40ac2cbbe9c57c31/raw/26617c60f33809cd5275799892878092f26cc1e9/hpc-test-node_W2) | :heavy_check_mark: SUCCESS
[hpc-test-node_W3](https://gist.githubusercontent.com/mellanox-github/bcc6a4599067962b40ac2cbbe9c57c31/raw/3cf1de744b74bd36d608823e22cf7db5d9b854b4/hpc-test-node_W3) | :heavy_check_mark: SUCCESS
[r-vmb-ppc-jenkins_W0](https://gist.githubusercontent.com/mellanox-github/bcc6a4599067962b40ac2cbbe9c57c31/raw/41eb8a9c60242e15b4ed3e4cec3c43a08de729f3/r-vmb-ppc-jenkins_W0) | :heavy_check_mark: SUCCESS
[r-vmb-ppc-jenkins_W1](https://gist.githubusercontent.com/mellanox-github/bcc6a4599067962b40ac2cbbe9c57c31/raw/e148cf926860dad8d8c2bbf9337ff8d5e9668a82/r-vmb-ppc-jenkins_W1) | :heavy_check_mark: SUCCESS
[r-vmb-ppc-jenkins_W2](https://gist.githubusercontent.com/mellanox-github/bcc6a4599067962b40ac2cbbe9c57c31/raw/d890dec27a75f9efc3509cb8fe7dd518be889218/r-vmb-ppc-jenkins_W2) | :heavy_check_mark: SUCCESS
[r-vmb-ppc-jenkins_W3](https://gist.githubusercontent.com/mellanox-github/bcc6a4599067962b40ac2cbbe9c57c31/raw/a7be1010d3032df6f3123924b48e4db74faf11af/r-vmb-ppc-jenkins_W3) | :heavy_check_mark: SUCCESS
[vulcan-jenkins_W0](https://gist.githubusercontent.com/mellanox-github/bcc6a4599067962b40ac2cbbe9c57c31/raw/cbb003881c70c8c71446baf53b88b5425f62a295/vulcan-jenkins_W0) | :heavy_check_mark: SUCCESS
[vulcan-jenkins_W1](https://gist.githubusercontent.com/mellanox-github/bcc6a4599067962b40ac2cbbe9c57c31/raw/c1f1ea562046f1b490ac574c30ff688a48b54798/vulcan-jenkins_W1) | :heavy_check_mark: SUCCESS
[vulcan-jenkins_W2](https://gist.githubusercontent.com/mellanox-github/bcc6a4599067962b40ac2cbbe9c57c31/raw/f1460620eb4ae8ad3ebab9920905e6182665142c/vulcan-jenkins_W2) | :heavy_check_mark: SUCCESS
[vulcan-jenkins_W3](https://gist.githubusercontent.com/mellanox-github/bcc6a4599067962b40ac2cbbe9c57c31/raw/428bc5491b12d6acfd29c8f496f9480ce82def79/vulcan-jenkins_W3) | :heavy_check_mark: SUCCESS

**Pull request merged**